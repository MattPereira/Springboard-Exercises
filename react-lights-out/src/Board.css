.Board {
  display: flex;
  justify-content: center;
  margin: 3rem;
}

.winner {
  font-size: 10rem;
  color: seagreen;
}