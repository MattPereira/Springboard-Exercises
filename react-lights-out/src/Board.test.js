import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Board from "./Board";

describe("<Board /> rendering", function () {
  describe("initial board and win state", function () {
    /* Smoke test*/
    it("renders without crashing", function () {
      render(<Board />);
    });

    /* Snapshot test*/
    it("matches snapshot for all lights on", function () {
      const { asFragment } = render(<Board chanceLightStartsOn={1} />);
      expect(asFragment()).toMatchSnapshot();
    });

    it("shows win state when all lights off", function () {
      render(<Board chanceLightStartsOn={0} />);
      expect(screen.getByText("You Won!")).toBeInTheDocument();
    });
  });
});

describe("cell click", function () {
  it("toggles lights properly", function () {
    //render a fully lit up board
    render(<Board nrows={3} ncols={3} chanceLightStartsOn={1} />);

    const cells = screen.getAllByRole("button");
    console.log(cells);

    //Expecting all cells start as lit
    cells.forEach((cell) => {
      expect(cell).toHaveClass("Cell-lit");
    });

    //[0,1,2]
    //[3,4,5]
    //[6,7,8]
    //click on the middle cell
    fireEvent.click(cells[4]);

    //Only cells at the corners should be lit
    let litIndices = [0, 2, 6, 8];
    cells.forEach((cell, idx) => {
      if (litIndices.includes(idx)) {
        expect(cell).toHaveClass("Cell-lit");
      } else {
        expect(cell).not.toHaveClass("Cell-lit");
      }
    });
  });

  it("says you won after turning all lights off", function () {
    render(<Board nrows={1} ncols={3} chanceLightStartsOn={1} />);

    expect(screen.queryByText("You Won!")).not.toBeInTheDocument();

    const winningCell = screen.getByText("0-1");
    fireEvent.click(winningCell);
    expect(screen.queryByText("You Won!")).toBeInTheDocument();
  });
});
/* test for clicking on a cell*/
