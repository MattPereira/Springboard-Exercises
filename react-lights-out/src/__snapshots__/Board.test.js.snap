// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> rendering initial board and win state matches snapshot for all lights on 1`] = `
<DocumentFragment>
  <div
    class="Board"
  >
    <table>
      <tbody>
        <tr>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            0-0
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            0-1
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            0-2
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            0-3
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            0-4
          </td>
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            1-0
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            1-1
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            1-2
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            1-3
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            1-4
          </td>
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            2-0
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            2-1
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            2-2
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            2-3
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            2-4
          </td>
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            3-0
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            3-1
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            3-2
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            3-3
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            3-4
          </td>
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            4-0
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            4-1
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            4-2
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            4-3
          </td>
          <td
            class="Cell Cell-lit"
            role="button"
          >
            4-4
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
